console.go

```diff
@@ -401,21 +401,38 @@ func consoleDefaultFormatTimestamp(timeFormat string, location *time.Location, n
 }
 
 func consoleDefaultFormatLevel(noColor bool) Formatter {
+	prefix := func(s string) string {
+		if len(s) > 3 {
+			return s[:3]
+		}
+		return s
+	}
 	return func(i interface{}) string {
+		const unknownLevel = "???"
+
 		var l string
 		if ll, ok := i.(string); ok {
 			level, _ := ParseLevel(ll)
 			fl, ok := FormattedLevels[level]
 			if ok {
 				l = colorize(fl, LevelColors[level], noColor)
 			} else {
-				l = strings.ToUpper(ll)[0:3]
+				if len(ll) == 0 {
+					l = unknownLevel
+				} else {
+					l = strings.ToUpper(prefix(ll))
+				}
```

Comment:
What about making all this logic a function instead of just prefix?

Reply:
Done :)